File: src/api/ai-analysis.ts
import { api } from '@/lib/api';

// 类型定义
export interface Topic {
  label: string;
  sentiment: 'positive' | 'negative' | 'neutral';
  count: number;
  trend: 'up' | 'down' | 'stable';
}

export interface TagCluster {
  category: string;
  items: string[];
  percentage: number;
  color: string;
}

export interface SentimentSummary {
  score: number;
  trend: string;
  positive: number;
  negative: number;
  aiAccuracy?: number;
  ai_accuracy?: number;
}

export interface RiskLevels {
  high: { count: number; desc: string };
  medium: { count: number; desc: string };
  low: { count: number; desc: string };
}

export interface ReplyRecord {
  id: string;
  user: string;
  rating: number;
  time: string;
  content: string;
  reply: string;
  status: string;
}

export interface ReplyStats {
  todayCount: number;
  autoBlocked: number;
  avgTime: string;
  total?: number;
  ai_generated?: number;
  manual?: number;
  success_rate?: number;
}

export interface AppealSuggestion {
  id?: string;
  review_id: string;
  user?: string;
  platform?: string;
  date?: string;
  content: string;
  draft?: string;
  is_malicious?: boolean;
  confidence?: number;
  suggestion?: string;
  appeal_content?: string;
}

// ========== API 调用 ==========

export const fetchTopics = async (period?: string): Promise<Topic[]> => {
  const url = period ? `/v1/ai-analysis/topics?period=${period}` : '/v1/ai-analysis/topics';
  const response = await api.get<any>(url);
  return response.data || [];
};

export const fetchTagClustering = async (period?: string): Promise<TagCluster[]> => {
  const url = period ? `/v1/ai-analysis/tag-clustering?period=${period}` : '/v1/ai-analysis/tag-clustering';
  const response = await api.get<any>(url);
  return response.data || [];
};

export const fetchSentimentSummary = async (): Promise<SentimentSummary> => {
  const response = await api.get<any>('/v1/ai-analysis/sentiment-summary');
  const data = response.data || response;
  return {
    score: data.score ?? 0,
    trend: data.trend ?? '-',
    positive: data.positive ?? 0,
    negative: data.negative ?? 0,
    aiAccuracy: data.aiAccuracy ?? data.ai_accuracy ?? 0,
  };
};

export const fetchRiskLevels = async (): Promise<RiskLevels> => {
  const response = await api.get<any>('/v1/ai-analysis/risk-levels');
  const data = response.data || response;
  return {
    high: { count: data.high_count ?? 0, desc: data.high_desc ?? '需立即处理' },
    medium: { count: data.medium_count ?? 0, desc: data.medium_desc ?? '建议24小时内回复' },
    low: { count: data.low_count ?? 0, desc: data.low_desc ?? '常规回复即可' },
  };
};

export const fetchReplyHistory = async (page = 1, pageSize = 20): Promise<ReplyRecord[]> => {
  const response = await api.get<any>(`/v1/ai-analysis/reply-history?page=${page}&page_size=${pageSize}`);
  const items = response.items || response.data || [];
  return items.map((item: any) => ({
    id: item.id || item.review_id || '',
    user: item.user_name || item.user || '',
    rating: item.rating || 0,
    time: item.created_at || item.time || '',
    content: item.content || '',
    reply: item.reply || item.ai_reply_content || '',
    status: item.status === 'completed' || item.status === 'sent' ? '自动已发' : item.status === 'approved' ? '已审核' : '待发送',
  }));
};

export const fetchReplyStats = async (): Promise<ReplyStats> => {
  const response = await api.get<any>('/v1/ai-analysis/reply-stats');
  const data = response.data || response;
  return {
    todayCount: data.total ?? 0,
    autoBlocked: data.ai_generated ?? 0,
    avgTime: '-',
    total: data.total ?? 0,
    ai_generated: data.ai_generated ?? 0,
    manual: data.manual ?? 0,
    success_rate: data.success_rate ?? 0,
  };
};

export const fetchAppealSuggestions = async (): Promise<AppealSuggestion[]> => {
  const response = await api.get<any>('/v1/ai-analysis/appeal-suggestions');
  const data = response.data || [];
  return Array.isArray(data) ? data : [];
};

File: backend/app/schemas/ai_analysis.py
"""
AI分析模块Schema定义
用于移动端AI分析相关接口的请求和响应模型
"""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class TopicResponse(BaseModel):
    """语义分析主题响应模型"""

    label: str = Field(..., description="主题标签")
    sentiment: str = Field(..., description="情感倾向: positive/negative/neutral")
    count: int = Field(..., description="提及次数")
    trend: str = Field(..., description="趋势: up/down/stable")


class TagClusterResponse(BaseModel):
    """差评标签聚类响应模型"""

    category: str = Field(..., description="标签分类")
    items: list[str] = Field(..., description="标签项列表")
    percentage: float = Field(..., description="占比百分比")
    color: str = Field(..., description="显示颜色")


class SentimentSummaryResponse(BaseModel):
    """情感指数响应模型"""

    score: float = Field(..., description="情感得分(0-100)")
    trend: str = Field(..., description="趋势: up/down/stable")
    positive: int = Field(..., description="正面评价数")
    negative: int = Field(..., description="负面评价数")
    ai_accuracy: float = Field(..., description="AI识别准确率")


class RiskLevelsResponse(BaseModel):
    """风险分级响应模型"""

    high_count: int = Field(..., description="高风险数量")
    high_desc: str = Field(..., description="高风险描述")
    medium_count: int = Field(..., description="中风险数量")
    medium_desc: str = Field(..., description="中风险描述")
    low_count: int = Field(..., description="低风险数量")
    low_desc: str = Field(..., description="低风险描述")


class ReplyHistoryResponse(BaseModel):
    """自动回复历史响应模型"""

    id: UUID = Field(..., description="记录ID")
    review_id: UUID = Field(..., description="评论ID")
    content: str = Field(..., description="回复内容")
    ai_generated: bool = Field(..., description="是否AI生成")
    status: str = Field(..., description="状态: pending/approved/rejected/sent")
    created_at: datetime = Field(..., description="创建时间")


class ReplyStatsResponse(BaseModel):
    """回复统计响应模型"""

    total: int = Field(..., description="总回复数")
    ai_generated: int = Field(..., description="AI生成数")
    manual: int = Field(..., description="手动回复数")
    success_rate: float = Field(..., description="成功率")


class AppealSuggestionResponse(BaseModel):
    """申诉建议响应模型"""

    review_id: UUID = Field(..., description="评论ID")
    is_malicious: bool = Field(..., description="是否为恶意差评")
    confidence: float = Field(..., description="置信度(0-1)")
    suggestion: str = Field(..., description="处理建议")
    appeal_content: str = Field(..., description="申诉文案")
